civiform/cloud-deploy-infra
# Add support for pgadmin container.

Terraform implementation for https://github.com/civiform/civiform/issues/2563.

New resources are disabled by default by using count = 0 on the module.

This will be followed up with an additional PR that sets count = 1 via a python command wrapper.

Tested:
1. Set count to 1, built a new docker image, ran `USE_DOCKER=true bin/deploy --tag=latest` in my config repo.
2. Terraform successfully deployed the pgadmin resources.
3. I navigated to https://avritt-dev.civiform.dev:4433 and successfully logged in using the created username and password.
4. I opened the CiviForm db group which prompted for the database password.  I successfully authenticated to the postgres database.
5. I successfully queried the postgres database with `SELECT * FROM questions` which returned the 2 default questions.
6. Set the count to 0, built a new docker image, ran `USE_DOCKER=true bin/deploy --tag=latest` in my config repo.
7. Terraform successfully destroyed the pgadmin resources.
8. I manually checked the lb listener rules, target groups, ecs services, ecs tasks, and secrets in the AWS UI to ensure the resources were properly deleted.
